packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/e2e/pageObject.ts

```diff
@@ -115,148 +121,219 @@ export class GeneratedAnswerObject {
-      (data: Record<string, any>, customData: Record<string, any>) => {
-        return (
-          Object.keys(expectedFields).every(
-            (key) => data?.[key] === expectedFields[key]
-          ) &&
-          Object.keys(expectedCustomFields).every(
-            (key) => customData?.[key] === expectedCustomFields[key]
+    if (this.analyticsMode === AnalyticsModeEnum.legacy) {
+      return this.analytics.waitForCustomUaAnalytics(
+        {
+          eventType: 'generatedAnswer',
+          eventValue: 'generatedAnswerSourceHover',
+        },
+        (event) =>
+          event?.customData?.generativeQuestionAnsweringId === this.streamId &&
+          Object.keys(expectedPayload).every(
+            (key) => event?.customData?.[key] === expectedPayload[key]
           )
-        );
-      }
+      );
+    }
+    return this.analytics.waitForEventProtocolAnalytics(
+      'Rga.CitationHover',
+      (event) =>
+        event.citationId === expectedPayload.citationId &&
+        event.itemMetadata?.uniqueFieldName === 'permanentid' &&
+        event.itemMetadata?.uniqueFieldValue === expectedPayload.permanentId
     );
   }
 
-  async waitForGeneratedAnswerClickUaAnalytics(
-    actionCause: string,
-    customChecker?: Function
+  async waitForFeedbackSubmitAnalytics(
+    expectedPayload: Record<string, any>
   ): Promise<Request> {
-    const uaRequest = this.page.waitForRequest((request) => {
-      if (isUaClickEvent(request)) {
-        const requestBody = request.postDataJSON?.();
-        const requestData = JSON.parse(requestBody.clickEvent);
-
-        const expectedFields: Record<string, any> = {
-          actionCause,
-        };
-
-        const matchesExpectedFields = Object.keys(expectedFields).every(
-          (key) => requestData?.[key] === expectedFields[key]
-        );
-
-        const customData = requestData?.customData;
-
-        const matchesGenerativeId =
-          customData?.generativeQuestionAnsweringId === this.streamId;
+    if (this.analyticsMode === AnalyticsModeEnum.legacy) {
+      return this.analytics.waitForCustomUaAnalytics(
+        {
+          eventType: 'generatedAnswer',
+          eventValue: 'generatedAnswerFeedbackSubmitV2',
+        },
+        (event) =>
+          event?.customData?.generativeQuestionAnsweringId === this.streamId &&
+          Object.keys(expectedPayload).every(
+            (key) => event?.customData?.[key] === expectedPayload[key]
+          )
+      );
+    }
 
-        return (
-          matchesExpectedFields &&
-          matchesGenerativeId &&
-          (customChecker ? customChecker(requestData, customData) : true)
-        );
-      }
-      return false;
-    });
-    return uaRequest;
+    function removeUnknownFields(object: Record<string, unknown>) {
+      return Object.fromEntries(
+        Object.entries(object).filter(([, value]) => value !== 'unknown')
+      );
+    }
+    const payloadToMatch = removeUnknownFields(expectedPayload);
```

Comment:
Typically this is how we create an inline function:

```suggestion
    const removeUnknownFields = (object: Record<string, unknown>) => {
      return Object.fromEntries(
        Object.entries(object).filter(([, value]) => value !== 'unknown')
      );
    }
    const payloadToMatch = removeUnknownFields(expectedPayload);
```

But I think better yet, we probably shouldn't declare the function at all and just use it like so:

```
const payloadToMatch = Object.fromEntries(
  Object.entries(expectedPayload).filter(([_, value]) => value !== 'unknown')
);
```

But also why do we have "unknown" fields? Is it the fields that the user has not completed? We don't explicitely send "unknown"? We just omit them in the EP event?